Makefile

```diff
@@ -332,6 +349,8 @@ precommit: prerequisites
 	$(Q)+$(MAKE) --no-print-directory clean
 	$(Q) echo "Running checks..."
 	$(Q)+$(MAKE) --no-print-directory check-signed-off check-vera check-cppcheck
+	$(Q) echo "...testing build-options..."
+	$(Q)+$(MAKE) --no-print-directory test-buildoptions
```

Comment:
It should be discussed and decided whether we want this to be included in the precommit target or have a separate target for the "heavy" checks (as was the case in a previous version, if I recall correctly). What's the impact of this addition on the precommit time? (Ofc, it can also be a valid answer that we don't care and want this tested, whatever it takes.)